pkg/alertmanager/alerts/store.go

```diff
@@ -0,0 +1,64 @@
+package alerts
+
+import (
+	"context"
+
+	"github.com/cortexproject/cortex/pkg/configs"
+	"github.com/cortexproject/cortex/pkg/configs/client"
+)
+
+// AlertStore stores and configures users rule configs
+type AlertStore interface {
+	ListAlertConfigs(ctx context.Context) (map[string]AlertConfigDesc, error)
+}
+
+// ConfigAlertStore is a concrete implementation of RuleStore that sources rules from the config service
+type ConfigAlertStore struct {
+	configClient client.Client
+	since        configs.ID
+	alertConfigs map[string]AlertConfigDesc
```

Comment:
Why do we need to retain it? Looking at `ListAlertConfigs()` implementation looks not useful to me.

Reply:
`configs, err := c.configClient.GetAlerts(ctx, c.since)`

Will only return configs that have `IDs` greater than `c.since`. This avoids the configdb having to do a full pull of it's table.